src/components/skills-quiz/data/utils.jsx

```diff
@@ -50,18 +51,27 @@ export function linkToCourse(course, slug, enterpriseUUID) {
   if (!Object.keys(course).length) {
     return '#';
   }
-  const queryParams = new URLSearchParams();
+  // If the course type has a path slug configured, add it to the url
+  const courseTypeConfig = getCourseTypeConfig(course);
+  const slugPlusCourseType = courseTypeConfig?.pathSlug ? `${slug}/${courseTypeConfig.pathSlug}` : slug;
+  const baseUrl = `/${slugPlusCourseType}/course/${course.key}`;
+  let query = '';
   if (course.queryId && course.objectId) {
+    const queryParams = new URLSearchParams();
     queryParams.set('queryId', course.queryId);
     queryParams.set('objectId', course.objectId);
+    query = `?${queryParams.toString()}`;
   }
-  const { userId } = getAuthenticatedUser();
-  sendEnterpriseTrackEvent(
-    enterpriseUUID,
-    'edx.ui.enterprise.learner_portal.skills_quiz.course.clicked',
-    { userId, enterprise: slug, selectedCourse: course.key },
-  );
-  return `/${slug}/course/${course.key}?${queryParams.toString()}`;
+
+  if (enterpriseUUID) {
+    const { userId } = getAuthenticatedUser();
+    sendEnterpriseTrackEvent(
+      enterpriseUUID,
+      'edx.ui.enterprise.learner_portal.skills_quiz.course.clicked',
+      { userId, enterprise: slug, selectedCourse: course.key },
+    );
```

Comment:
Previously, this function was only called inside of a click handler within the skills quiz feature, where it made sense to include a Segment event like this.

However, through this PR, we're now essentially re-purposing this function as a generic utility outside of the skills quiz and outside of a click handler. That said, we probably don't want to be dispatching a Segment event intended to track clicks at time of generating the URL to be clicked. Should this Segment event be dispatched outside of this function, where relevant, instead?

Also, this Segment event is specific to the skills quiz (i.e., `edx.ui.enterprise.learner_portal.skills_quiz.course.clicked`) and the function is now being used outside of the skills quiz feature.

Reply:
Good call-out, moved out of the function and into the one spot in the code it was used.